stackstate_checks_base/stackstate_checks/base/utils/http_helper.py

```diff
@@ -0,0 +1,1447 @@
+import json
+import requests_mock
+from enum import Enum
+from schematics.models import Model
+from schematics.types import StringType
+from schematics.exceptions import DataError
+from requests import Session, Request
+from requests.auth import HTTPBasicAuth
+from requests.adapters import HTTPAdapter
+from requests.packages.urllib3.util.retry import Retry
+
+try:
+    import urlparse
+except ImportError:
+    import urllib.parse as urlparse
+
+
+"""
+    Authentication Models
+"""
+
+
+class HTTPBasicAuthentication(Model):
+    username = StringType(required=True)
+    password = StringType(required=True)
+
+
+"""
+    Authentication ENUM
+"""
```

Comment:
I am a bit puzzled by your docstrings. AFAIK you have to put the docstring underneath the class/function declaration.  E.g.

```
class HTTPAuthenticationType(Enum):
    """
    Authentication ENUM
    """
```

Reply:
Sorry I'm use to defining information above the starting line for the defined information